### PR TITLE
feature/MainScreen

### PR DESCRIPTION
**Pull request by nhanviet1**
## Related Tickets
- [#Task: Create Navigation for MainScreen](https://edu-redmine.sun-asterisk.vn/issues/xxxx)
## WHAT
- Create Navigation for MainScreen
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/formula1/view/SplashActivity.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        lifecycleScope.launch(job) {
+            delay(DELAY_TIME)
+            startActivity(Intent(this@SplashActivity, MainScreen::class.java))
+            finish()
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
```
Comment (hungld-1302) [inline app/src/main/java/com/example/formula1/view/SplashActivity.kt]: job.cancel
super.onDestroy()

File: app/src/main/java/com/example/formula1/view/mainscreen/MainScreen.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+                circuitId -> {
+                    if (!circuitState) {
+                        fm.beginTransaction().add(R.id.fragment_holder, circuitFragment)
+                            .show(circuitFragment).hide(standingFragment).hide(raceFragment)
+                            .commit()
+                        circuitState = true
+                    } else {
+                        fm.beginTransaction().show(circuitFragment).hide(standingFragment)
+                            .hide(raceFragment).commit()
```
Comment (hunggld) [inline app/src/main/java/com/example/formula1/view/mainscreen/MainScreen.kt:55]: viết addFragmet vào extension

File: app/src/main/java/com/example/formula1/utils/ViewExtension.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.example.formula1.utils
+
+import android.view.View
+
+fun View.isGone() {
```
Comment (minhnn-1279) [inline app/src/main/java/com/example/formula1/utils/ViewExtension.kt]:
```suggestion
fun View.gone() {
```

Comment (minhnn-1279) [reply app/src/main/java/com/example/formula1/utils/ViewExtension.kt]: mấy cái hàm kia cx bỏ is nhé :v nó ko phải là check điều kiện ý

File: app/src/main/res/color/selected_item_nav_color.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:state_selected="true" android:color="@color/color_red" />
+    <item android:color="@android:color/black"  />
+</selector>
```
Comment (hungld-1302) [inline app/src/main/res/color/selected_item_nav_color.xml]: endline

File: app/src/main/res/layout/activity_main_screen.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    <com.google.android.material.bottomnavigation.BottomNavigationView
+        android:id="@+id/bottom_nav"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:itemIconTint="@color/selected_item_nav_color"
+        app:itemTextColor="@color/selected_item_nav_color"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:menu="@menu/menu" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (hungld-1302) [inline app/src/main/res/layout/activity_main_screen.xml]: .

File: app/src/main/res/layout/fragment_circuits.xml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    tools:context=".view.mainscreen.fragments.CircuitsFragment">
+
+    <!-- TODO: Update blank fragment layout -->
+    <TextView
+        android:id="@+id/textTest"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:text="circuits" />
+
+</FrameLayout>
```
Comment (hungld-1302) [inline app/src/main/res/layout/fragment_circuits.xml]: .

File: app/src/main/res/layout/fragment_racing.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    android:layout_height="match_parent"
+    tools:context=".view.mainscreen.fragments.RacingFragment">
+
+    <!-- TODO: Update blank fragment layout -->
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:text="racing" />
+
+</FrameLayout>
```
Comment (hungld-1302) [inline app/src/main/res/layout/fragment_racing.xml]: .

File: app/src/main/res/layout/fragment_standing.xml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    android:layout_height="match_parent"
+    tools:context=".view.mainscreen.fragments.StandingFragment">
+
+    <!-- TODO: Update blank fragment layout -->
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:text="standing" />
+
+</FrameLayout>
```
Comment (hungld-1302) [inline app/src/main/res/layout/fragment_standing.xml]: .

File: app/src/main/res/menu/menu.xml [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+        android:icon="@drawable/ic_checkered_flag"/>
+    <item
+        android:id="@+id/standing"
+        android:title="Standings"
+        android:icon="@drawable/ic_trophy"/>
+    <item
+        android:id="@+id/circuit"
+        android:title="Circuits"
+        android:icon="@drawable/ic_circuit"/>
+</menu>
```
Comment (hungld-1302) [inline app/src/main/res/menu/menu.xml]: .

File: app/src/main/java/com/example/formula1/view/SplashActivity.kt [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        animation.repeatCount = Animation.INFINITE
+        animation.repeatMode = Animation.REVERSE
+        binding.imgLogo.startAnimation(animation)
+
+        lifecycleScope.launch(job) {
+            delay(DELAY_TIME)
+            startActivity(Intent(this@SplashActivity, MainScreen::class.java))
+            finish()
+        }
+    }
```
Comment (minhnn-1279) [inline app/src/main/java/com/example/formula1/view/SplashActivity.kt]: ông truyền job vào làm gì nhỉ :v

File: app/src/main/res/color/selected_item_nav_color.xml (hunk 4 above)
Comment (minhnn-1279) [inline app/src/main/res/color/selected_item_nav_color.xml]: endline

File: app/src/main/java/com/example/formula1/view/SplashActivity.kt (hunk 10 above)
Comment (nhanviet1) [reply app/src/main/java/com/example/formula1/view/SplashActivity.kt:57]: Để dưới tôi cancel cái Job thì cái coroutine scope nó đc hủy á

File: app/src/main/java/com/example/formula1/utils/ViewExtension.kt [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package com.example.formula1.utils
+
+import android.view.View
+
+fun View.gone() {
+    visibility = View.GONE
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/formula1/utils/ViewExtension.kt]: if(visibility != View.GONE) { visibility = View.GONE }
tương tự ở dưới nhé

**Pull request merged**